### ISSUE TITLE
Missing idMSO's for Excel's context menus

### ISSUE THREAD

**Issue by kpuls** [OfficeDev/office-fluent-ui-command-identifiers#15]
Documentation is missing as to the idMso's on some of Excel's context menus.  Once specific case is ContextMenuCell (this context menu is listed in Control Name column, but no child commands are not listed at all.)  I have been able to successfully identify the idMSOs for Cut and Copy, but many others such as Insert/Delete/PasteSpecial do not work. It would be really nice to get these idMSOs listed.

**Comment by Rick-Kirkham** [comment]
I'm trying to find the answer to this internally.

**Comment by LGDes** [comment]
Waiting for this from 2014 :-)
[https://answers.microsoft.com/es-es/msoffice/forum/all/saber-todos-los-nombres-de-los-comandos-botones/38c3bf3a-d46c-4ccb-aebc-1c1906eb535f]

**Comment by LGDes** [comment]
> Waiting for this since 2014 :-) [https://answers.microsoft.com/es-es/msoffice/forum/all/saber-todos-los-nombres-de-los-comandos-botones/38c3bf3a-d46c-4ccb-aebc-1c1906eb535f]

any news on this?

**Comment by LGDes** [comment]
still no news? :(

**Comment by Rick-Kirkham** [comment]
@LGDes I'm sorry for the repeated delays. I've just escalated this to a higher authority and requested a response this week.

**Comment by LGDes** [comment]
> @LGDes I'm sorry for the repeated delays. I've just escalated this to a higher authority and requested a response this week.

Happy new year... still no news? (more than 1 week ;D)

**Comment by Rick-Kirkham** [comment]
@LGDes Sigh. I just pinged the email thread about this again. This time I added a mild scolding and said that I'm embarrassed on behalf of Microsoft that this is taking so long. I will be on vacation for the next 3 weeks, but I will ask my manager to follow up if there isn't a prompt response.

**Comment by Rick-Kirkham** [comment]
This is what I'm told: Some menus in Office applications are dynamically generated at runtime, including the one for ContextMenuCell. The item controls on these menus are not exposed to 3rd party developers. So, you're not going to be able to reference them in your applications/extensions. 

Again, I apologize for how long it took to get this information to you. I will leave this issue open for a while in case there are further developments.

**Comment by kpuls** [comment]
<sigh>  
Rick, I appreciate your efforts in running this to ground to get an answer.  So please be aware that this is a general comment, not directed at you personally...

It is pretty frustrating to hear that the "extensibility" framework cannot be used for extensibility.  It is also pretty frustrating that the rules do not seem to be applied consistently here.

I mean, I could almost understand this if the menus were dynamically adding a variable number of elements based on active content - say a list of open workbooks or something.  But I cannot, for even a minute, understand why the idMSO for the PasteSpecial button cannot be provided.  There is an IdMso for this already.  The docs show it lives on both the Home Tab and the ContextMenuChartMenu (at a minimum).  Did someone use a different idMSO for the CellContextMenu?  If so, why on earth would they do that?  Doesn't that defeat a major philosophy point of the entire "fluent ui" extensibility model?  I'm not asking about commands unique to these menus.  I'm asking about commands that exist elsewhere in the app with the exact same names and existing idMSOs.

Look at the GetData menu as a model of how this is supposed to work. Some of those commands get hidden based on your office SKU.  But I can get at the idMso of every single one.  That is how things should be working for the Cell Context Menu as well.  And it if isn't - I would suggest that this is actually a software bug, not just a "lack of documentation" bug.

Please note that I have not done any further testing on this since I raised the issue almost 8 months ago, but iirc, the PasteSpecial menu item cannot be controlled via the PastSpecialDialog idMSO for the cell context menu like it can everywhere else in the app.  Dynamically generated or not, it absolutely should.

**Comment by LGDes** [comment]
> It is pretty frustrating to hear that the "extensibility" framework cannot be used for extensibility. It is also pretty frustrating that the rules do not seem to be applied consistently here.
> 
> Did someone use a different idMSO for the CellContextMenu? If so, why on earth would they do that? Doesn't that defeat a major philosophy point of the entire "fluent ui" extensibility model?
>
> I would suggest that this is actually a software bug, not just a "lack of documentation" bug.
> 

Totally agree!!!

**Comment by jkpieterse** [comment]
I agree as well. I'm trying to disable the refresh button on the Queries & connections Task pane right-click menu, so far I have not found a way to do that.

**Comment by LGDes** [comment]
Seems need to wait for another 10 years to get a 'real' solution...
